common/lib/dependabot/pull_request_creator/bitbucket.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require "dependabot/clients/bitbucket"
+require "dependabot/pull_request_creator"
+
+module Dependabot
+  class PullRequestCreator
+    class Bitbucket
+      attr_reader :source, :branch_name, :base_commit, :credentials,
+                  :files, :commit_message, :pr_description, :pr_name,
+                  :author_details, :labeler, :work_item
+
+      def initialize(source:, branch_name:, base_commit:, credentials:,
+                     files:, commit_message:, pr_description:, pr_name:,
+                     author_details:, labeler:, work_item: nil)
+        @source         = source
+        @branch_name    = branch_name
+        @base_commit    = base_commit
+        @credentials    = credentials
+        @files          = files
+        @commit_message = commit_message
+        @pr_description = pr_description
+        @pr_name        = pr_name
+        @author_details = author_details
+        @labeler        = labeler
+        @work_item      = work_item
+      end
+
+      def create
+        return if branch_exists? && pull_request_exists?
+
+        # FIXME: Copied from Azure, but not verified whether this is true
+        # For Bitbucket we create or update a branch in the same request as creating
+        # a commit (so we don't need create or update branch logic here)
+        create_commit
+
+        create_pull_request
+      end
+
+      private
+
+      # FIXME: I don't know what the "for_source" methods are for
+      def bitbucket_client_for_source
+        @bitbucket_client_for_source ||=
+          Dependabot::Clients::Bitbucket.for_source(
+            source: source,
+            credentials: credentials
+          )
+      end
+
+      def branch_exists?
+        bitbucket_client_for_source.branch(source.repo, branch_name)
```

Comment:
Nit: it's weird that we passed `source` to the constructor for `Dependabot::Clients::Bitbucket.for_source`, but still have to pass attributes like `source.repo` to individual methods.

If `source` isn't used by `Dependabot::Clients::Bitbucket` it may be cleanest to remove.